packages/browser/src/plugins/env-enrichment/index.ts

```diff
@@ -195,32 +140,13 @@ class PageEnrichmentPlugin implements Plugin {
   }
 
   private enrich = (ctx: Context): Context => {
-    const event = ctx.event
-    const evtCtx = (event.context ??= {})
-
-    const defaultPageContext = pageDefaults()
-
-    let pageContextFromEventProps: Pick<EventProperties, string> | undefined
+    // Note: Types are off - context should never be undefined here, since it is set as part of event creation.
+    const evtCtx = ctx.event.context!
 
-    if (event.type === 'page') {
-      pageContextFromEventProps =
-        event.properties &&
-        pick(event.properties, Object.keys(defaultPageContext))
-
-      event.properties = {
-        ...defaultPageContext,
-        ...event.properties,
-        ...(event.name ? { name: event.name } : {}),
-      }
-    }
-
-    evtCtx.page = {
-      ...defaultPageContext,
-      ...pageContextFromEventProps,
-      ...evtCtx.page,
-    }
+    const search = evtCtx.page!.search || ''
 
-    const query: string = evtCtx.page.search || ''
+    const query =
+      typeof search === 'object' ? objectToQueryString(search) : search
```

Comment:
In what scenario is `search` and object?

Reply:
Users sometimes override search with dictionary that contains parsed search params (e.g they use NextJS where it's on the router object), generally when they are making a page call.

Reply:
I was wondering how we'd support query strings that had multiple values for the same key.  (Wasn't sure what was making the object in the first place for it to be a concern) For example `?foo=a&foo=b`

Looks like next allows strings or string arrays for the value:
```
NextApiRequest.query: Partial<{
    [key: string]: string | string[];
}>
```
But `new URLSearchParam()` won't work with `{ 'foo': ['a', 'b'] }` - it'd have to look like `[['foo', 'a'], ['foo', 'b']]`. It does accept `Record<string, string>`, but treats `Record<string, string[]>` oddly.

Reply:
We could probably convert the object straight to the array of arrays version easily enough.

Reply:
I wasn't suggesting we convert anything to  URLSearchParams, I was saying that we it would be plausible  accept that as an argument and convert them to a string here, as I'm doing with object.

Reply:
@chrisradek I updated the code to handle  `[key: string]: string | string[]`.